Foundation/Collection/Foldable.hs

```diff
@@ -43,6 +45,20 @@ class Foldable collection where
     foldr' :: (Element collection -> a -> a) -> a -> collection -> a
     foldr' f z0 xs = foldl f' id xs z0 where f' k x z = k $! f x z
 
+-- | Fold1's. Like folds, but they assume to operate on a NonEmpty collection.
+class Foldable f => Fold1able f where
+    -- | Left associative lazy fold.
+    foldl1  :: (Element f -> Element f -> Element f) -> NonEmpty f -> Element f
```

Comment:
we are actually considering removing the lazy `foldl` ( #319). Might be worth removing the lazy `foldl1` too. What do you think ?